src/test/java/suite/optout/OptoutTest.java

```diff
@@ -99,25 +101,88 @@ public void testV2LogoutWithV2TokenGenerate(String label, Operator operator, Str
+        assumeThat(tokenGenerateVersion).isEqualTo("v0");
+
+        JsonNode response = this.waitForOptOutResponse(operator::v0CheckedTokenRefresh, refreshToken, "{\"advertisement_token\":\"\",\"advertising_token\":\"\",\"refresh_token\":\"\"}");
+
+        assertThat(response).isEqualTo(Mapper.OBJECT_MAPPER.readTree("{\"advertisement_token\":\"\",\"advertising_token\":\"\",\"refresh_token\":\"\"}"));
+    }
+
+    @Order(6)
+    @ParameterizedTest(name = "/v1/token/refresh after {3} generate and {4} logout - {0} - {2}")
+    @MethodSource({
+            "newRefreshTokenArgs"
+    })
+    public void testV1TokenRefreshAfterOptOut(String label, Operator operator, String tokenGenerateVersion, String tokenLogoutVersion, String refreshToken, String refreshResponseKey) throws Exception {
+        assumeThat(tokenGenerateVersion).isEqualTo("v1");
+
+        JsonNode response = this.waitForOptOutResponse(operator::v1CheckedTokenRefresh, refreshToken, "{\"status\":\"optout\"}");
+
+        assertThat(response).isEqualTo(Mapper.OBJECT_MAPPER.readTree("{\"status\":\"optout\"}"));
+    }
+
+    @Order(7)
+    @ParameterizedTest(name = "/v2/token/refresh after {3} generate and {4} logout - {0} - {2}")
+    @MethodSource({
+            "newRefreshTokenArgs"
+    })
+    public void testV2TokenRefreshAfterOptOut(String label, Operator operator, String tokenGenerateVersion, String tokenLogoutVersion, String refreshToken, String refreshResponseKey) throws Exception {
+        assumeThat(tokenGenerateVersion).isEqualTo("v2");
+
+        JsonNode response = operator.v2TokenRefresh(refreshToken, refreshResponseKey);
+        int breakCounter = 0;
+        while (breakCounter < OPTOUT_WAIT_SECONDS && !response.equals(Mapper.OBJECT_MAPPER.readTree("{\"status\":\"optout\"}"))) {
+            TimeUnit.SECONDS.sleep(5);
+            response = operator.v2TokenRefresh(refreshToken, refreshResponseKey);
+            breakCounter++;
+        }
+
+        assertThat(breakCounter).isLessThan(OPTOUT_WAIT_SECONDS).withFailMessage("Timed-out getting an Opt Out Response");
+        assertThat(response).isEqualTo(Mapper.OBJECT_MAPPER.readTree("{\"status\":\"optout\"}"));
+    }
+
+    private static Set<Arguments> newRefreshTokenArgs() {
+        return outputArgs;
+    }
+
+    private void addToken(String label, Operator operator, String tokenGenerateVersion, String tokenLogoutVersion, String refreshToken, String refreshResponseKey) {
+        outputArgs.add(Arguments.of(label, operator, tokenGenerateVersion, tokenLogoutVersion, refreshToken, refreshResponseKey));
+    }
+
+    private JsonNode waitForOptOutResponse(Function<String, JsonNode> tokenRefreshFunction, String refreshToken, String expectedResponse) {
+        try {
+            JsonNode response = tokenRefreshFunction.apply(refreshToken);
+            int breakCounter = 0;
+            while (breakCounter < OPTOUT_WAIT_SECONDS && !response.equals(Mapper.OBJECT_MAPPER.readTree(expectedResponse))) {
```

Comment:
Maybe we could use something like https://github.com/awaitility/awaitility to wait instead of `Thread.sleep()`?

Reply:
Added